### PR TITLE
[TRACE] Add trace info for all kinds of task

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -390,6 +391,7 @@ void TaskWorkerPool::_drop_tablet_worker_thread_callback() {
         TDropTabletReq drop_tablet_req;
         {
             lock_guard<Mutex> worker_thread_lock(_worker_thread_lock);
+            TRACE("get worker thread lock");
```
Comment (acelyc111) [inline be/src/agent/task_worker_pool.cpp:394]: Same as above

File: be/src/agent/task_worker_pool.cpp [hunk 2]
```diff
@@ -402,6 +404,17 @@ void TaskWorkerPool::_drop_tablet_worker_thread_callback() {
...
         }
 
+        scoped_refptr<Trace> trace(new Trace);
+        MonotonicStopWatch watch;
+        watch.start();
+        SCOPED_CLEANUP({
+            if (watch.elapsed_time() / 1e9 > config::agent_task_trace_threshold_sec) {
+                LOG(WARNING) << "Trace:" << std::endl << trace->DumpToString(Trace::INCLUDE_ALL);
+            }
+        });
```
Comment (acelyc111) [inline be/src/agent/task_worker_pool.cpp:414]: Better to define a macro for these code, it's duplicate with code in ceate tablet, and if you add more traces for other tasks, there will be manu duplicate code.

File: be/src/agent/task_worker_pool.cpp [hunk 3]
```diff
@@ -402,6 +404,17 @@ void TaskWorkerPool::_drop_tablet_worker_thread_callback() {
...
+        ADOPT_TRACE(trace.get());
+        
+        TRACE("begin to drop tablet");
```
Comment (acelyc111) [inline be/src/agent/task_worker_pool.cpp:417]: How about add another TRACE after line 422 when tablet got?

File: be/src/agent/task_worker_pool.cpp (hunk 2 above)
Comment (xinghuayu007) [reply be/src/agent/task_worker_pool.cpp]: update

**Review by acelyc111 — DISMISSED** [review]

**Review by morningman — COMMENTED** [review]
The default `agent_task_trace_threshold_sec` is only 2 seconds,
which may lead to a lot of tracing dump log.

Maybe we can modify the default value to a large number.

**Comment by xinghuayu007** [comment]
> The default `agent_task_trace_threshold_sec` is only 2 seconds,
> which may lead to a lot of tracing dump log.
> 
> Maybe we can modify the default value to a large number.

how about set to 10s?

**Comment by xinghuayu007** [comment]
> The default `agent_task_trace_threshold_sec` is only 2 seconds,
> which may lead to a lot of tracing dump log.
> 
> Maybe we can modify the default value to a large number.

`agent_task_trace_threshold_sec` is modified to 86400 seconds (one day). Anyone who wants to open trace feathure, just config it lower.

**Comment by acelyc111** [comment]
> > The default `agent_task_trace_threshold_sec` is only 2 seconds,
> > which may lead to a lot of tracing dump log.
> > Maybe we can modify the default value to a large number.
> 
> `agent_task_trace_threshold_sec` is modified to 86400 seconds (one day). Anyone who wants to open trace feathure, just config it lower.

How about set default to -1 which means disable it?

**Comment by xinghuayu007** [comment]
> > > The default `agent_task_trace_threshold_sec` is only 2 seconds,
> > > which may lead to a lot of tracing dump log.
> > > Maybe we can modify the default value to a large number.
> > 
> > 
> > `agent_task_trace_threshold_sec` is modified to 86400 seconds (one day). Anyone who wants to open trace feathure, just config it lower.
> 
> How about set default to -1 which means disable it?

good idea.

**Pull request closed without merging**